src/goto-cc/gcc_mode.cpp

```diff
@@ -655,7 +656,7 @@ int gcc_modet::doit()
 
   // We now iterate over any input files
 
-  temp_dirt temp_dir("goto-cc-XXXXXX");
+  std::forward_list<temp_dirt> temp_dirs;
```

Comment:
Why not just a vector?

Reply:
This was done since `temp_dirt` deletes the temporary dir upon destruction. So when the vector elements are relocated the directories would be deleted. Though thinking about it the proper fix would be to add a move constructor to `temp_dirt` and delete the copy constructor.